typings/index.d.ts

```diff
@@ -228,3 +228,136 @@ export type DownshiftInterface<Item> = React.ComponentClass<
 declare const Downshift: DownshiftInterface<any>
 export default Downshift
 export function resetIdCounter(): void
+
+/* useSelect Types */
+
+export interface UseSelectState<Item> {
+  highlightedIndex: number
+  selectedItem: Item
+  isOpen: boolean
+  keySoFar: string
+}
+
+export enum UseSelectStateChangeTypes {
+  MenuKeyDownArrowDown = '__menu_keydown_arrow_down__',
+  MenuKeyDownArrowUp = '__menu_keydown_arrow_up__',
+  MenuKeyDownEscape = '__menu_keydown_escape__',
+  MenuKeyDownHome = '__menu_keydown_home__',
+  MenuKeyDownEnd = '__menu_keydown_end__',
+  MenuKeyDownEnter = '__menu_keydown_enter__',
+  MenuKeyDownCharacter = '__menu_keydown_character__',
+  MenuBlur = '__menu_blur__',
+  ItemMouseMove = '__item_mouse_move__',
+  ItemClick = '__item_click__',
+  ToggleButtonKeyDownCharacter = '__togglebutton_keydown_character__',
+  ToggleButtonKeyDownArrowDown = '__togglebutton_keydown_arrow_down__',
+  ToggleButtonKeyDownArrowUp = '__togglebutton_keydown_arrow_up__',
+  ToggleButtonClick = '__togglebutton_click__',
+  FunctionToggleMenu = '__function_toggle_menu__',
+  FunctionOpenMenu = '__function_open_menu__',
+  FunctionCloseMenu = '__function_close_menu__',
+  FunctionSetHighlightedIndex = '__function_set_highlighted_index__',
+  FunctionSelectItem = '__function_select_item__',
+  FunctionClearKeysSoFar = '__function_clear_keys_so_far__',
+  FunctionReset = '__function_reset__',
+}
+
+export interface UseSelectProps<Item> {
+  items: Item[]
```

Comment:
I think this could actually be `readonly Item[]` as Downshift doesn't mutate this array, right? What do you think about changing it?